File: tests/fixtures/hello.ts
const name: string = Deno.env.get("USER") ?? "world";

interface Greeting {
  message: string;
  count: number;
}

function greet(name: string, times: number): Greeting[] {
  return Array.from({ length: times }, (_, i) => ({
    message: `hello ${name}!`,
    count: i + 1,
  }));
}

for (const g of greet(name, 3)) {
  console.log(`${g.message} (${g.count})`);
}

File: tests/fixtures/hello.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	name := os.Getenv("USER")
	if name == "" {
		name = "world"
	}
	greeting := fmt.Sprintf("hello %s!", strings.Title(name))
	fmt.Println(greeting)
}
